packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -366,6 +382,19 @@ export class Dialog
   // #endregion
 
   // #region Private Methods
+
+  private handleFocusTrapDisabled(): void {
+    if (!this.open) {
+      return;
+    }
+
+    if (this._focusTrapDisabledOverride()) {
```

Comment:
This check should be moved inside `activateFocusTrap`. Within that function, if `focusTrapDisabledOverride` is defined, it should be called.